### PR TITLE
Check previous outputs of burns/inputs duplicated in tx 

### PR DESCRIPTION
**Pull request by jhs7jhs**
Resolves: #517

### CODE REVIEW

File: types/src/transaction/transaction.rs [hunk 1]
```diff
@@ -492,6 +493,20 @@ impl HeapSizeOf for Transaction {
     }
 }
 
+fn is_prev_out_duplicated(burns: &[AssetTransferInput], inputs: &[AssetTransferInput]) -> Result<(), Error> {
```
Comment (sgkim126) [inline types/src/transaction/transaction.rs]: How about renaming it to `verify_duplicated_prev_out`?
`is_*` seems to return a boolean.

File: types/src/transaction/transaction.rs [hunk 2]
```diff
@@ -492,6 +493,20 @@ impl HeapSizeOf for Transaction {
...
+    let mut prev_out_set = HashSet::new();
+    for input in inputs.iter().chain(burns) {
+        let prev_out = (input.prev_out.transaction_hash, input.prev_out.index);
+        if !prev_out_set.insert(prev_out) {
+            return Err(Error::DuplicatedPreviousOutput {
+                transaction_hash: input.prev_out.transaction_hash.clone(),
```
Comment (sgkim126) [inline types/src/transaction/transaction.rs]: `H256` derives `Copy`. No need to call `clone`.

File: types/src/transaction/transaction.rs (hunk 1 above)
Comment (jhs7jhs) [reply types/src/transaction/transaction.rs]: Changed to `check_duplication_in_prev_out`.

File: types/src/transaction/transaction.rs (hunk 2 above)
Comment (jhs7jhs) [reply types/src/transaction/transaction.rs]: Fixed.

**Pull request merged**